feeds/scheduler/feed_group.go

```diff
@@ -125,14 +124,21 @@ func (fg *FeedGroup) publishPackages(pkgs []*feeds.Package) (int, error) {
 		}).Print("Sending package upstream")
 		b, err := json.Marshal(pkg)
 		if err != nil {
-			log.Printf("Error marshaling package: %#v", pkg)
-			return processed, err
+			log.WithField("name", pkg.Name).WithError(err).Error("Error marshaling package")
+			errs = append(errs, err)
 		}
 		if err := (fg.publisher).Send(context.Background(), b); err != nil {
-			log.Printf("Error sending package to upstream publisher %v", err)
-			return processed, err
+			log.WithField("name", pkg.Name).WithError(err).Error("Error sending package to upstream publisher")
```

Comment:
The only side effect of this is that it could flood the logs with repeated identical error messages 
```
ERRO[0027] Error sending package to upstream publisher   error="pubsub (code=Unknown): kafka: Failed to deliver 1 messages." name=@skylla/rep
INFO[0027] Sending package upstream                      created_date="2021-06-15 12:49:18.2 +0000 UTC" feed=npm name=@docusaurus/utils-validation
ERRO[0027] Error sending package to upstream publisher   error="pubsub (code=Unknown): kafka: Failed to deliver 1 messages." name=@docusaurus/utils-validation
INFO[0027] Sending package upstream                      created_date="2021-06-15 12:49:12.177 +0000 UTC" feed=npm name=@shack-js/template-repo
ERRO[0027] Error sending package to upstream publisher   error="pubsub (code=Unknown): kafka: Failed to deliver 1 messages." name=@shack-js/template-repo
```
Not a blocking comment, just a thought

Reply:
Yeh fair point, the names of the packages which failed to be sent could definitely be stored and then logged in a single entry, or grouped by error type. For now I think this is fine though.